sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,235 @@
+    must align with parent axis after the rotation.
+
+    Parameters
+    ==========
+
+    name : string
+        The Joint's name which makes it unique.
+    parent : Body
+        The parent body of joint.
+    child : Body
+        The child body of joint.
+    coordinates: dynamicsymbol, optional
+        Coordinates of joint.
+    speeds : dynamicsymbol, optional
+        Speeds of joint.
+    parent_joint_pos : Vector, optional
+        Defines the joint's point where the parent will be connected to child.
+        Default value is masscenter of Parent Body.
+    child_joint_pos : Vector, optional
+        Defines the joint's point where the child will be connected to parent.
+        Default value is masscenter of Child Body.
+    parent_axis : Vector, optional
+        Axis of parent frame which would be be aligned with child's
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of Child frame which would be be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Examples
+    =========
+
+    Adding a Pin Joint which connects center of mass of parent to a point
+    pointed by child.frame.x + child.frame.y w.r.t. child's center of mass.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import Body, PinJoint
+    >>> parent = Body('parent')
+    >>> child = Body('child')
+    >>> P = PinJoint('P', parent, child)
+    >>> P.coordinates()
+    [P_theta(t)]
+    >>> P.speeds()
+    [P_omega(t)]
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis = None):
+
+        super().__init__(name, parent, child, coordinates, speeds, parent_joint_pos, child_joint_pos,
+            parent_axis)
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            theta = dynamicsymbols(self._name + '_theta')
+            coordinates.append(theta)
+            return coordinates
+        coordinates.append(coordinate)
+        return coordinates
+
+    def _generate_speeds(self, speed):
+        speeds = []
+        if speed is None:
+            omega = dynamicsymbols(self._name + '_omega')
+            speeds.append(omega)
+            return speeds
+        speeds.append(speed)
+        return speeds
+
+    def _orient_frames(self):
+        self._child.frame.orient_axis(self._parent.frame, self._parent_axis, self._coordinates[0])
+
+    def _set_angular_velocity(self):
+        self._child.frame.set_ang_vel(self._parent.frame, self._speeds[0] * self._parent_axis)
+
+    def _set_linear_velocity(self):
+        self._parent_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_pos(self._parent_joint, 0)
+        self._child.masscenter.v2pt_theory(self._parent.masscenter, self._parent.frame, self._child.frame)
```

Comment:
@moorepants s I think everything is working fine except this function. The linear velocity is being calculated incorrectly.